Doc/library/sqlite3.rst

```diff
@@ -2270,13 +2344,72 @@ can be found in the `SQLite URI documentation`_.
 Explanation
 -----------
 
+.. _sqlite3-transaction-control:
 .. _sqlite3-controlling-transactions:
 
 Transaction control
 ^^^^^^^^^^^^^^^^^^^
 
-The :mod:`!sqlite3` module does not adhere to the transaction handling recommended
-by :pep:`249`.
+:mod:`!sqlite3` offers multiple methods of controlling whether,
+when and how database transactions are opened and closed.
+:ref:`sqlite3-transaction-control-autocommit` is recommended,
+while :ref:`sqlite3-transaction-control-isolation-level`
+retains the pre-Python 3.12 behaviour.
+
+.. _sqlite3-transaction-control-autocommit:
+
+Transaction control via the ``autocommit`` attribute
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The recommended way of controlling transaction behaviour is through
+the :attr:`Connection.autocommit` attribute,
+which should preferrably be set using the *autocommit* parameter
+of :func:`connect`.
+
+It is suggested to set *autocommit* to ``False``,
+which implies :pep:`249`-compliant transaction control.
+This means:
+
+* :mod:`!sqlite3` ensures that a transaction is always open,
+  so :meth:`Connection.commit` and :meth:`Connection.rollback`
+  will implicitly open a new transaction immediately after closing
+  the pending one.
+  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
+* Transactions should be committed explicitly using :meth:`!commit`.
+* Transactions should be rolled back explicitly using :meth:`!rollback`.
+* An implicit rollback is performed if the database is
+  :meth:`~Connection.close`-ed with pending changes.
```

Comment:
```suggestion
It is suggested to set *autocommit* to ``False``,
which implies :pep:`249`-compliant transaction control.
This means:

* :mod:`!sqlite3` ensures that a transaction is always open,
  so :meth:`Connection.connect`, :meth:`Connection.commit`,
  and :meth:`Connection.rollback` implicitly open a new transaction;
  transactions should not be opened explicitly with a `BEGIN` statement.
  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
* Transactions should be committed explicitly using :meth:`!commit`;
  transactions should not be committed explicitly with a `COMMIT` statement.
* Transactions should be rolled back explicitly using :meth:`!rollback`;
  transactions should not be rolled back explicitly with a `ROLL BACK` statement.
* :meth:`Connection.close` implicitly rolls back any pending transaction.
```

Reply:
IMO, this makes the text more unreadable and fails to clearly communicate the effects of `autocommit is False`.

Reply:
I compared your text with [@malemburg’s specification](https://github.com/python/cpython/issues/83638#issuecomment-1093853924) for this paragraph:

> If this is set to False, the module could then implement the
> correct way of handling transactions, which means:
> 
> a) start a new transaction when the connection is opened
> b) start a new transaction after .commit() and .rollback()
> c) don't start new transactions anywhere else
> d) run an implicit .rollback() when the connection closes

and I failed to see the items a) and c) in your text.